beets/library.py

```diff
@@ -49,7 +50,7 @@
 # Library-specific query types.
 
 
-class SingletonQuery(dbcore.FieldQuery):
+class SingletonQuery(dbcore.FieldQuery[NoReturn]):
```

Comment:
Shouldn't this rather be `str`?

It's a weird one: I've written this class defining `__new__` method with the same parameters like `__init__` has in the base class. It's essentially a 'factory' class that returns an instance of `NotQuery`.

`mypy` won't be able to map the defined to the parameter in `__new__` definition, but since we expect it to have a string there, it may be worth defining it so?

Reply:
Oh, I didn't realize that `SingletonQuery` can be used to search for `album_id is None` as well as `album_id is not None` (and even then, `()` would have been a better fit than `NoReturn`, I suppose).  Since we're just calling `util.str2bool()` on the provided pattern, maybe we should just take a `bool` there.  What do you think?

Reply:
It looks like `SingletonQuery` gets pulled in to `beets.dbcore.db.Model._queries` (of type `Dict[str, Type[FieldQuery[Any]]]`), with the documentation "Named queries that use a field-like `name:value` syntax".  In that case, I understand why `SingletonQuery` would take `str` instead of `bool` -- but shouldn't every query in `_queries` also be of type `FieldQuery[str]`?  Is there any reason for the pattern to be a generic parameter at all?

Reply:
> Since we're just calling `util.str2bool()` on the provided pattern, maybe we should just take a `bool` there. What do you think?

For this exact reason, I thought `str` would be more suitable, given that it's the string input that is being converted to boolean?

Reply:
> but shouldn't every query in _queries also be of type FieldQuery[str]? Is there any reason for the pattern to be a generic parameter at all?

The `FieldQuery` being constructed from the input (string) query is only one of the uses cases - in other cases it's actually initialised with the correct type, so it needs to be generic, as far as I remember.

Reply:
Alright, I see.  I'm going to keep it as `str` then.  However, I am concerned with the use of `util.str2bool()`: at some point, it should be changed to only accept unambiguously positive or negative answers (i.e. it should be able to fail), to avoid confusion for users, but that's out of scope for this PR.